### ISSUE TITLE
Method SearchUsingPost doesn't work when SearchParams has encoded values 

### ISSUE THREAD

**Issue by DaniloRochaGR7** [FirelyTeam/fhir-net-api#1519]
### Description
I have subscriptions with encoded values on the criteria, and I need to check they exist using the exact same criteria.

### Expected Behavior
When I try to search for a subscription using `SearchUsingPost` and I have the `SearchParams` with encoded values on the criteria, and the subscription is returned on a bundle.

### Current Behavior
When I try to search for a subscription using `SearchUsingPost` and I have the `SearchParams` with encoded values, then the request can't find the subscription. If I use some criteria where the value is not encoded, the subscription is returned as expected.

e.g: 

```c#
searchParams.Add("criteria", "Observation?patient.identifier=http://somehost/fhir/Name%20Hospital|someId");

```
### Steps to Reproduce (for bugs)
- Create a subscription with an encoded value;
- Use a SearchUsingPost method to try retrieving the subscription;
- The bundle will return empty;

### Your Environment
I'm using Hl7.Fhir.STU3.

### Version used:
- 1.6.0
- 1.9.0

**Comment by marcovisserFurore** [comment]
Thanks @KillerGR7 for creating this issue. I can confirm that this is indeed a bug. The following unittest proves that:

```c#
        [TestMethod]
        public async Task SearchUsingPost_Issue1591()
        {
            using var client = new FhirClient(_endpoint);
            client.Settings.PreferredFormat = ResourceFormat.Json;
            client.Settings.PreferredReturn = Prefer.ReturnRepresentation;

            var srch = new SearchParams()
                       .Where(@"criteria=Observation?patient.identifier=http://somehost/fhir/Name%20Hospital|someId");

            var result1 = await client.SearchUsingPostAsync<Subscription>(srch);
            Assert.IsTrue(result1.Entry.Count >= 1);
        }
```

When I do the same thing in Postman (Search using POST) then I am able to find the encoded value.

**Comment by kennethmyhra** [comment]
@KillerGR7, @marcovisserFurore and me have tested against a few servers
* Vonk - https://vonk.fire.ly/r3/
* HAPI - http://hapi.fhir.org/baseDstu3/
* Spark - https://spark-stu3.incendi.no/fhir/
* Health Intersections FHIR Server - http://test.fhir.org/r3

We can make it work against both Vonk and Spark, but are having trouble against HAPI and Health Intersections. We have also looked through our code and we can't see anything that could possibly mess up. We handle form-url encoded parameters using the [FormUrlEncodedContent](https://docs.microsoft.com/en-us/dotnet/api/system.net.http.formurlencodedcontent?view=netstandard-2.0) class.

The only way I could make it work against HAPI was if I changed `%20` to `+`, see more information about URL encoding white spaces here: https://en.wikipedia.org/wiki/Query_string#URL_encoding

Let us now if you have more information

**Comment by DaniloRochaGR7** [comment]
Thank you for the feedback!

Based on what you described, can I assume that this is a matter of how the HAPI was implemented and not the FHIR net API?

I tried to search using a `+` sign, but it also didn't work, anyway, I'm using a HAPI FHIR server fed by an external source, that's where the encoded values come from, so unfortunately the way they're implemented is out of my hands, I was able to parse the criteria though, using the `ParseQueryString` splitting the criteria and adding `criteria: contains` to the search params, this way I can check each of the parameters individually and avoid the encoded values, once I get the resource bundle I can double-check if I have the correct subscription.
